nixos/modules/services/web-apps/engelsystem.nix

```
@@ -0,0 +1,171 @@
+{ config, lib, pkgs, utils, ... }:
+
+let
+  inherit (lib) mkDefault mkEnableOption mkIf mkOption types literalExample;
+
+  cfg = config.services.engelsystem;
+  fpm = config.services.phpfpm.pools.engelsystem;
+
+  genConfigScript = pkgs.writeScript "engelsystem-gen-config.sh" (utils.genJqSecretsReplacementSnippet cfg.config "config.json");
+in {
+  options = {
+    services.engelsystem = {
+      enable = mkEnableOption "Online tool for coordinating helpers and shifts on large events";
+
+      domain = mkOption {
+        type = types.str;
+        example = "engelsystem.example.com";
+        description = "domain to serve on";
+      };
+
+      package = mkOption {
+        type = types.package;
+        example = literalExample "pkgs.engelsystem";
+        description = "Engelsystem package used for the service";
+        default = pkgs.engelsystem;
+      };
+
+      createDatabase = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether to create a local database automatically.
+          This will overrid every database setting in <option>services.engelsystem.config</option>
+        '';
+      };
+    };
+
+    services.engelsystem.config = mkOption {
+      type = types.attrs;
+      default = {
+        database = {
+          unix_socket = "/var/run/mysqld/mysqld.sock";
+          host = "localhost";
+          database = "engelsystem";
+          username = "engelsystem";
+        };
+      };
+      example = {
+        email = {
+          maintnance = false;
+          driver = "smtp";
+          from.address = "engelsystem@example.com";
+          from.name = "example engesyste";
+          encryption = "tls";
+          username = "engelsystem@example.com";
+          password._secret = "/var/keys/engelsystem/mail";
+        };
+        autoarrive = true;
+        min_password_length = 6;
+        default_locale = "de_DE";
+      };
```

I would suggest the following, which includes an example of external database configuration, fixes the `maintenance` option, and includes an smtp `host` and `port`.
```suggestion
      example = {
        maintenance = false;
        database = {
          host = "database.example.com";
          database = "engelsystem";
          username = "engelsystem";
          password._secret = "/var/keys/engelsystem/database";
        };
        email = {
          driver = "smtp";
          host = "smtp.example.com";
          port = 587;
          from.address = "engelsystem@example.com";
          from.name = "example engelsystem";
          encryption = "tls";
          username = "engelsystem@example.com";
          password._secret = "/var/keys/engelsystem/mail";
        };
        autoarrive = true;
        min_password_length = 6;
        default_locale = "de_DE";
      };
```

---

I think this is a good example, will use them